packages/cache/src/AzureBlobCacheStorage.ts

```diff
@@ -29,41 +31,65 @@ function createBlobClient(
 }
 
 export class AzureBlobCacheStorage extends CacheStorage {
-  constructor(private options: AzureBlobCacheStorageOptions) {
+  constructor(
+    private options: AzureBlobCacheStorageOptions,
+    private internalCacheFolder: string
+  ) {
     super();
   }
 
-  protected _fetch(hash: string, destinationFolder: string): Promise<boolean> {
-    const blobClient = createBlobClient(
-      this.options.connectionString,
-      this.options.container,
+  protected async _fetch(hash: string): Promise<string | undefined> {
+    const temporaryBlobOutputFolder = path.join(
```

Comment:
Why do we need an temporary folder?

Reply:
And not directly the package folder?

Reply:
Is it to satisfy the abstraction of the mother class? If so maybe the abstraction of the mother can be refactored so it does not forces the implementations to do extra work.

Reply:
Let's benchmark and change later. Right now it is a decent abstraction that forces the storage provider implementations to behave similarly, and it provides a local cache tier.